framework/apache-superset/conftest.py

```diff
@@ -0,0 +1,131 @@
+import os
+import shlex
+import shutil
+import subprocess
+import time
+
+import pytest
+import requests
+
+from util import get_auth_headers
+
+
+superset_env = {
+    "FLASK_APP": "superset",
+    "SUPERSET_CONFIG_PATH": "superset_config.py",
+}
+superset_bin = shutil.which("superset")
+
+
+uri_database = "http://localhost:8088/api/v1/database/"
+
+
+# Utility functions.
+
+def invoke_superset(command: str):
+    """
+    Invoke `superset` command.
+    """
+    command = f"{superset_bin} {command}"
+    subprocess.check_call(shlex.split(command), env=superset_env, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+
+# Test suite fixtures.
+
+@pytest.fixture(scope="session")
+def fix_greenlet():
+    """
+    Install more recent greenlet, because Playwright installs version 3.0.1, which breaks Superset.
+    """
+    os.system("pip install --upgrade greenlet")
```

Comment:
Well. Irgendwas ist immer. May need a report or patch on/for Playwright.

Reply:
There is a corresponding patch now.

- https://github.com/microsoft/playwright-python/pull/2227